experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxyNodeCursor.ts

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+import { type Value, type FieldKey, EmptyKey, TreeTypeSet, rootFieldKey } from "../../../core";
+// eslint-disable-next-line import/no-internal-modules
+import { leaf } from "../../../domains/leafDomain";
+import {
+	ContextuallyTypedNodeData,
+	TreeDataContext,
+	getPossibleTypes,
+	isFluidHandle,
+	isPrimitiveValue,
+} from "../../contextuallyTyped";
+import {
+	type CursorAdapter,
+	type CursorWithNode,
+	stackTreeNodeCursor,
+	stackTreeFieldCursor,
+} from "../../treeCursorUtils";
+import { type TreeNodeSchema } from "../../typed-schema";
+import { type ProxyNode } from "./types";
+
+/**
+ * Creates a {@link CursorAdaptor} capable of mapping proxy-domain trees to the editable tree domain.
+ *
+ * @param context - The {@link TreeEntity.context} of the parent node under which the tree is being inserted.
+ * @param typeSet - The types allowed under the parent node under which the tree is being inserted.
+ *
+ * @privateRemarks
+ * This is exported solely for testing purposes.
+ * Consumers should use {@link cursorFromProxyTreeNode} instead.
+ */
+export function createAdaptor<TNode extends ProxyNode<TreeNodeSchema, "javaScript">>(
+	context: TreeDataContext,
+	typeSet: TreeTypeSet,
+): CursorAdapter<TNode> {
+	return {
+		value: (node: TNode) => {
+			return isPrimitiveValue(node) || node === null || isFluidHandle(node)
+				? (node as Value)
+				: undefined;
+		},
+		type: (node: TNode) => {
+			if (node === undefined) {
+				throw new Error("undefined node");
```

Comment:
Is it possible to hit this? If the schema is `{ foo?: number }` and the user provides `{}`, then certainly this won't be hit. But what if they provide `{ foo: undefined }`? Do we handle that case? 

If we do, I would expect it to be handled in a layer above - whatever code is passing stuff to the cursor adaptor should filter out explicit undefineds. If that is the case, then this should probably be an `assert` rather than an Error (asserts are typically better for internal invariants being broken, whereas Errors are better for things a user might cause).

Reply:
I was hitting this while my code had bugs. I don't remember the exact scenario where it was happening though :/

Reply:
I wasn't able to repro again, but note that the ProxyNode type includes `unknown` which can allows for `undefined`. We don't expect `undefined` to be provided, so I think explicit error boundaries here is appropriate.